src/types.rs

```diff
@@ -921,3 +921,102 @@ impl Debug for Log {
+    /// See `TransactionReceiptKind` for summaries and references.
+    #[serde(flatten)]
+    pub kind: TransactionReceiptKind,
+    /// The hash of the transaction that emitted this log.
+    pub transaction_hash: Digest,
+    /// The index of the transaction that emitted this log within the block.
+    pub transaction_index: U256,
+    /// The hash of the block containing the transaction.
+    pub block_hash: Digest,
+    /// The height of the block containing the transaction.
+    pub block_number: U256,
+    /// Address of transaction sender.
+    pub from: Address,
+    /// Transaction receipient ([`None`] for contract creation).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to: Option<Address>,
+    /// The price paid post-execution by the transaction (i.e. base fee + priority fee).
+    /// Both fields in 1559-style transactions are *maximums* (max fee + max priority fee), the
+    /// amount that's actually paid by users can only be determined post-execution.
+    pub effective_gas_price: U256,
+    /// The sum of gas used by this transaction and all preceding transactions in the same block.
+    pub cumulative_gas_used: U256,
+    /// The amount of gas used for this specific transaction alone.
+    pub gas_used: U256,
+    /// Contract address created, or `None` if not a deployment.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub contract_address: Option<Address>,
+    /// Logs emitted by the transaction.
+    pub logs: Vec<Log>,
+    /// The log bloom filter.
+    pub logs_bloom: Bloom,
+    #[serde(flatten)]
+    pub status: TransactionReceiptStatus,
+}
+
+/// Status: either 1 (success) or 0 (failure).
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(tag = "status")]
+pub enum TransactionReceiptStatus {
+    #[serde(rename = "0x0")]
+    Failure,
+    #[serde(rename = "0x1")]
+    Success,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(tag = "type")]
+pub enum TransactionReceiptKind {
+    /// - Legacy: The original transaction type
+    #[serde(rename = "0x0")]
+    Legacy,
+    /// - EIP2930: Optional access lists
+    /// https://eips.ethereum.org/EIPS/eip-2930
+    #[serde(rename = "0x1")]
+    Eip2930,
+    /// EIP1559: Fee market change for ETH 1.0 chain
+    /// - https://eips.ethereum.org/EIPS/eip-1559
+    #[serde(rename = "0x2")]
+    Eip1559,
+    /// EIP4844: Shard Blob Transactions
+    /// - https://eips.ethereum.org/EIPS/eip-4844
+    #[serde(rename = "0x3")]
+    Eip4844 {
+        /// The amount of blob gas used for this specific transaction.
+        /// Only specified for blob transactions as defined by EIP-4844.
+        #[serde(rename = "blobGasUsed")]
+        blob_gas_used: U256,
+        /// The actual value per gas deducted from the sender's account for blob gas.
+        /// Only specified for blob transactions as defined by [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844).
```

Comment:
Since link is already in the variant, I would just:

```suggestion
        /// Only specified for blob transactions as defined by EIP-4844.
```